src/javascript/binary_japan/contract.js

```diff
@@ -1,82 +1,108 @@
-if(typeof JAPAN === 'function'){
-    var Contract = (function(){
-    	'use strict';
-        var open, close, contractDetails = [], periods={}, contractType = {};
+if(typeof is_japan === 'function'){
+    var Contract = (function () {
+        'use strict';
+
+        var contractDetails = {}, contractType = {}, periods = {},
+            open, close, form, barrier;
 
         var populate_periods = function (currentContract){
-        	if(!periods[currentContract.contract_category]){
-        		periods[currentContract.contract_category] = {};
-        	}
-
-        	if(!periods[currentContract.contract_category][currentContract.underlying_symbol]){
-        		periods[currentContract.contract_category][currentContract.underlying_symbol] = {};
-        	}
-
-        	var period = currentContract.trading_period.date_start.epoch+'_'+currentContract.trading_period.date_expiry.epoch;
-
-        	// console.log(period, currentContract.expiry_type,currentContract);
-            var d = new Date(currentContract.trading_period.date_start.epoch*1000);
-
-        	periods[currentContract.contract_category][currentContract.underlying_symbol][period] = {
-        		available_barriers: currentContract.available_barriers,
-        		barrier: currentContract.barrier,
-        		high_barrier: currentContract.high_barrier,
-        		low_barrier: currentContract.low_barrier,
-        		barriers: currentContract.barriers,
-        		// expiry_type: currentContract.expiry_type,
-        		date_start: currentContract.trading_period.date_start,
-        		date_expiry: currentContract.trading_period.date_expiry,
-        		duration: currentContract.trading_period.duration
-        	};
+            var currentCategory = currentContract['contract_category'];
+            if(!periods[currentCategory]){
+                periods[currentCategory] = {};
+            }
+
+            if(!periods[currentCategory][currentContract.underlying_symbol]){
+                periods[currentCategory][currentContract.underlying_symbol] = {};
+            }
+
+            var period = currentContract['trading_period']['date_start']['epoch']+'_'+currentContract['trading_period']['date_expiry']['epoch'];
+
+            periods[currentCategory][currentContract['underlying_symbol']][period] = {
+                available_barriers: currentContract['available_barriers'],
+                barrier: currentContract['barrier'],
+                high_barrier: currentContract['high_barrier'],
+                low_barrier: currentContract['low_barrier'],
+                barriers: currentContract['barriers'],
+                date_start: currentContract['trading_period']['date_start'],
+                date_expiry: currentContract['trading_period']['date_expiry'],
+                duration: currentContract['trading_period']['duration']
+            };
```

Comment:
Are you doing a full object copy here?
We should gradually start modernizing our code base. Object.assign is available in ES6, and we might start using it soon via polyfill.